programs/programs/policyengine/calculators/dependencies/member.py

```diff
@@ -449,3 +449,27 @@ class HeadStart(Member):
 
 class EarlyHeadStart(Member):
     field = "early_head_start"
+
+
+class IsFppMedicaidEligibleDependency(Member):
+    """
+    Dependency for is_medicaid_eligible used by IL Family Planning Program.
+
+    This checks if the member is enrolled in full-time Medicaid or other
+    full coverage state medical program, which makes them ineligible for FPP.
+    """
+
+    field = "is_medicaid_eligible"
+    dependencies = ("insurance",)
+
+    def value(self):
+        # Check if member has Medicaid or family_planning insurance coverage
+        if hasattr(self.member, "insurance"):
+            return self.member.insurance.has_insurance_types(("medicaid", "family_planning"), strict=False)
```

Comment:
Wondering about the `family_planning` aspect of this. It's listed as FPPE in the UI, so it should probably just limit the display of FPPE (and not HFS FPP).

Curious if it makes sense to use broader language (e.g. Family Planning Program) on the insurance tile though?